kong/plugins/ai-prompt-guard/schema.lua

```diff
@@ -0,0 +1,50 @@
+local typedefs = require "kong.db.schema.typedefs"
+
+return {
+  name = "ai-prompt-guard",
+  fields = {
+    { protocols = typedefs.protocols_http },
+    { config = {
+      type = "record",
+      fields = {
+          { allow_patterns = {
```

Comment:
Experience has taught us that operators often use regex to spot the presence or absence of literal strings. This is often the case because we have not given them any other option.
I suggest that you change the configuration interface to accept two kinds of intents for matching the content:
- detect the presence of a literal string
- regex-based match

This changes the performance characteristics significantly since regexes are much more expensive than literal string searches.